playwright/_impl/_set_input_files_helpers.py

```diff
@@ -34,81 +34,61 @@
         if context._channel._connection.is_remote:
             streams = []
-            for file in file_list:
-                assert isinstance(file, (str, Path))
+            for item in items:
+                assert isinstance(item, (str, Path))
+                last_modified_ms = int(os.path.getmtime(item) * 1000)
                 stream: WritableStream = from_channel(
                     await context._channel.send(
-                        "createTempFile", {"name": os.path.basename(file)}
+                        "createTempFile",
+                        {
+                            "name": os.path.basename(item),
+                            "lastModifiedMs": last_modified_ms,
+                        },
                     )
                 )
-                await stream.copy(file)
+                await stream.copy(item)
                 streams.append(stream._channel)
-            return InputFilesList(streams=streams, localPaths=None, files=None)
-        local_paths = []
-        for p in file_list:
-            assert isinstance(p, (str, Path))
-            local_paths.append(str(Path(p).absolute().resolve()))
-        return InputFilesList(streams=None, localPaths=local_paths, files=None)
+            return InputFilesList(streams=streams)
+        return InputFilesList(
+            localPaths=[
+                str(Path(cast(Union[str, Path], item)).absolute().resolve())
+                for item in items
+            ]
+        )
 
-    return InputFilesList(
-        streams=None, localPaths=None, files=await _normalize_file_payloads(files)
+    file_payload_exceeds_size_limit = (
+        sum([len(f.get("buffer", "")) for f in items if not isinstance(f, (str, Path))])
+        > SIZE_LIMIT_IN_BYTES
     )
+    if file_payload_exceeds_size_limit:
+        raise Error(
+            "Cannot set buffer larger than 50Mb, please write it to a file and pass its path instead."
+        )
 
-
-async def _normalize_file_payloads(
-    files: Union[str, Path, FilePayload, List[Union[str, Path]], List[FilePayload]]
-) -> List:
-    file_list = files if isinstance(files, list) else [files]
-    file_payloads: List = []
-    for item in file_list:
-        if isinstance(item, (str, Path)):
-            file_payloads.append(
-                {
-                    "name": os.path.basename(item),
-                    "buffer": base64.b64encode(await async_readfile(item)).decode(),
-                }
-            )
-        else:
-            file_payloads.append(
-                {
-                    "name": item["name"],
-                    "mimeType": item["mimeType"],
-                    "buffer": base64.b64encode(item["buffer"]).decode(),
-                }
-            )
-
-    return file_payloads
+    return InputFilesList(
+        payloads=[
+            {
+                "name": item["name"],
+                "mimeType": item["mimeType"],
+                "buffer": base64.b64encode(item["buffer"]).decode(),
```

Comment:
why is it encode than decode, is it just the python way to do base64 encoding?

Reply:
b64encode returns bytes which we need to decode to a str.